src/HivePlot/HivePlot.tsx

```diff
@@ -178,197 +598,208 @@ export class HivePlot extends Component<HivePlotProps, HivePlotState> {
-    const { activeIds } = this.props;
-    const { active } = this.state;
-    // If no there is nothing active, then everything is active.
-    if (!active && !activeIds!.length) {
-      return true;
-    }
-
-    // If this node is active because it is being hovered
-    if (active && active[`${type}-${index}`]) {
-      return true;
-    }
-    // If the ID matches one of the active IDs passed in the props
-    if (
-      !!activeIds!.length &&
-      !!nodeOrLink.id &&
-      activeIds!.includes(nodeOrLink.id)
-    ) {
-      return true;
-    }
-    return false;
-  }
+              {label.show && (
+                <HiveLabel
+                  index={i}
+                  text={a.label}
+                  label={label}
+                  outerRadius={outerRadius}
+                  angle={angle}
+                />
+              )}
+            </g>
+          ))}
+        </Fragment>
+      );
+    },
+    [axis, label]
+  );
+
+  const isActive = useCallback(
+    (nodeOrLink: Node | Link, index: number, type: 'node' | 'link') => {
+      // If no there is nothing active, then everything is active.
+      if (!active && !activeIds!.length) {
+        return true;
+      }
 
-  renderLinks({ angle, radius, domainColor }) {
-    const { links } = this.props;
-    return (
-      <Fragment>
-        {links.map((link, i) => {
-          return (
-            <HiveLink
-              key={`${link.value}-${i}`}
-              color={link.color || domainColor}
-              active={this.isActive(link, i, 'link')}
-              angle={angle}
+      // If this node is active because it is being hovered
+      if (active && active[`${type}-${index}`]) {
+        return true;
+      }
+      // If the ID matches one of the active IDs passed in the props
+      if (
+        !!activeIds!.length &&
+        !!nodeOrLink.id &&
+        activeIds!.includes(nodeOrLink.id)
+      ) {
+        return true;
+      }
+      return false;
+    },
+    [activeIds, active]
+  );
+
+  const renderLinks = useCallback(
+    ({ angle, radius, domainColor }) => {
+      return (
+        <Fragment>
+          {links.map((link, i) => {
+            return (
+              <HiveLink
+                key={`${link.value}-${i}`}
+                color={link.color || domainColor}
+                active={isActive(link, i, 'link')}
+                angle={angle}
+                radius={radius}
+                link={link}
+                onMouseOver={(event: MouseEvent) =>
+                  onLinkMouseOverLocal(link, event)
+                }
+                onMouseOut={(event: MouseEvent) =>
+                  onLinkMouseOutLocal(link, event)
+                }
+              />
+            );
+          })}
+        </Fragment>
+      );
+    },
+    [links]
+  );
+
+  const renderNodes = useCallback(
+    ({ angle, radius, domainColor }) => {
+      return (
+        <Fragment>
+          {nodes.map((node, i) => (
+            <HiveNode
```

Comment:
I personally try to have the fns just render a specific thing rather than loop in situations like this and put the loop in the root.

Reply:
I actually mimic the behaviour before. 
But I prefer it to be this way, so the component is more clear - different strokes.

Reply:
Ya - its not a big deal - just a personal preference. Happy to keep it.